packages/salesforce-adapter/src/filters/utils.ts

```diff
@@ -451,13 +483,34 @@ export const getWhereConditions = (
       ))
       .join(' AND ')
   })
-  return r
+  if (r.length === 0) {
+    return [complexConditionQuery]
+  }
+  return r.map(queryChunk => `${queryChunk}${(queryChunk.length > 0 && limitingConditionSets.length > 0) ? ' AND ' : ''}${complexConditionQuery}`)
 }
 
-export const conditionQueries = (query: string, conditionSets: Record<string,
-  string>[], maxQueryLen = MAX_QUERY_LENGTH): string[] => {
+export const conditionQueries = (
+  query: string,
+  conditionSets: ReadonlyArray<ReadonlyArray<SoqlQuery>>,
+  queryLimits = DefaultSoqlQueryLimits,
+): string[] => {
+  if (conditionSets.length === 0) {
+    return [query]
+  }
+  const exactConditionSets = conditionSets
+    .map(conditions => conditions
+      .filter(condition => condition.operator === 'IN')
+      .map(condition => [condition.fieldName, condition.value]))
+    .filter(conditions => conditions.length > 0)
+    .map(Object.fromEntries)
```

Comment:
1. I find this logic complicated due to the execution order, and inner filters.
2. **Object.fromEntries** is anti-typescript as it returns **Object** of type **object** which is implicitly **any**

Consider something like:
```suggestion
const toExactConditionSet = (conditionSet: ReadonlyArray<SoqlQuery>): Record<string, string> | undefined => {
    const exactConditions = conditionSet
      .filter(condition => condition.operator === 'IN')
      .reduce<Record<string, string>>((r, condition) => {
        r[condition.fieldName] = condition.value
        return r
      }, {})
    return Object.keys(exactConditions).length > 0 ? exactConditions : undefined
  }
  const exactConditionSets = conditionSets
    .map(toExactConditionSet)
    .filter(isDefined)
```

Reply:
Ack